src/deploy/index.js

```diff
@@ -107,8 +107,12 @@ var deploy = function (targetNames, options, customContext = {}) {
         _.each(context.hosting.deploys, function (deploy) {
           logger.info(clc.bold("Hosting URL:"), utils.addSubdomain(api.hostingOrigin, deploy.site));
         });
-        const versionNames = context.hosting.deploys.map((deploy) => deploy.version);
-        return { hosting: versionNames.length === 1 ? versionNames[0] : versionNames };
+        var siteDetails = {};
+        context.hosting.deploys.forEach((deploy) => {
+          var version = deploy.version.replace(`sites/${deploy.site}/versions/`, "");
+          siteDetails[deploy.site] = { version };
+        });
+        return { hosting: siteDetails };
```

Comment:
This change may also have an effect on the output of `deploy --json`. Could you check the before/after of the behavior there?

Reply:
It looks compatible. The only new field is `version`.

Before:

```
{
  "status": "success",
  "result": {
    "<site>": {
      "site": "<site>",
      "url": "https://<site>--slam-test-6161efgc.web.app",
      "expireTime": "2021-03-02T19:40:07.615661137Z"
    }
  }
}
```

After:

```
{
  "status": "success",
  "result": {
    "<site>": {
      "site": "<site>",
      "url": "https://<site>--slam-test-6161efgc.web.app",
      "version": "09ef5462c91e2bba",
      "expireTime": "2021-03-02T19:39:05.793378023Z"
    }
  }
}
```

Reply:
This makes sense to me because before this PR the return value of `deploy()` was not used anywhere:

https://github.com/firebase/firebase-tools/pull/3157/files#diff-90fa65dc2e075506e4e87cc1c6bee0c8a1e769440908e956179ad0fa4a79d999L130